sound/soc/sof/mediatek/mt8195/mt8195.c

```diff
@@ -316,6 +316,20 @@ static int mt8195_dsp_suspend(struct snd_sof_dev *sdev, u32 target_state)
 {
 	struct platform_device *pdev = container_of(sdev->dev, struct platform_device, dev);
 	int ret;
+	int reset_sw, dbg_pc, sleep_try = 2000, sleep_cnt = 0;
+
+	/* wait 2 second to wait dsp enter wait for interrupt */
+	do {
+		reset_sw = snd_sof_dsp_read(sdev, DSP_REG_BAR, DSP_RESET_SW);
+		mdelay(1);
+		sleep_cnt++;
```

Comment:
two seconds? that seems wild, and if that is true then maybe use a different delay to let the hardware do what it needs to do.